File: 01-Ejercicios-Clases/C8-09-08-23-Prepared-Stm/ProyectosJAVA/LP2SoftRRHHController/src/pe/edu/pucp/lp2soft/rrhh/mysql/AreaMySQL.java
package pe.edu.pucp.lp2soft.rrhh.mysql;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import java.util.ArrayList;
import pe.edu.pucp.lp2soft.config.DBManager;
import pe.edu.pucp.lp2soft.rrhh.dao.AreaDAO;
import pe.edu.pucp.lp2soft.rrhh.model.Area;
public class AreaMySQL implements AreaDAO{
    private Connection con;
    private Statement st;
    private String sql;
    private ResultSet rs;
    @Override
    public int insertar(Area area) {
        int resultado = 0;
        try{
            con = DBManager.getInstance().getConnection();
            st = con.createStatement();
            sql = "INSERT INTO area(nombre,activa) VALUES("
                    + "'" + area.getNombre() + "',1)";
            resultado = st.executeUpdate(sql);
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex)
            {System.out.println(ex.getMessage());}
        }
        return resultado;
    }

    @Override
    public int modificar(Area area) {
        int resultado = 0;
        try{
            con = DBManager.getInstance().getConnection();
            st = con.createStatement();
            sql = "UPDATE area SET nombre = '" 
                    + area.getNombre()+"' WHERE id_area = "
                    + area.getIdArea();
            resultado = st.executeUpdate(sql);
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex)
            {System.out.println(ex.getMessage());}
        }
        return resultado;
    }

    @Override
    public int eliminar(int idArea) {
        int resultado = 0;
        try{
            con = DBManager.getInstance().getConnection();
            st = con.createStatement();
            sql = "UPDATE area SET activa = 0 WHERE id_area = "
                    + idArea;
            resultado = st.executeUpdate(sql);
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex)
            {System.out.println(ex.getMessage());}
        }
        return resultado;
    }

    @Override
    public ArrayList<Area> listarTodas() {
        ArrayList<Area> areas = new ArrayList<>();
        try{
            con = DBManager.getInstance().getConnection();
            st = con.createStatement();
            sql = "SELECT id_area,nombre "
                    + "FROM area WHERE activa = 1";
            rs = st.executeQuery(sql);
            while(rs.next()){
                Area area = new Area();
                area.setIdArea(rs.getInt("id_area"));
                area.setNombre(rs.getString("nombre"));
                area.setActivo(true);
                areas.add(area);
            }
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }finally{
            try{con.close();}catch(Exception ex){
                System.out.println(ex.getMessage());
            }
        }
        return areas;
    }
    
}

File: 01-Ejercicios-Clases/Proyecto-Curso/ProyectosNET/LP2Soft/LP2SoftRRHHController/MySQL/AreaMySQL.cs
using LP2SoftDBManager;
using LP2SoftRRHHController.DAO;
using LP2SoftRRHHModel;
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LP2SoftRRHHController.MySQL
{
    public class AreaMySQL : AreaDAO
    {
        private MySqlConnection con;
        // similar al CallableStatement
        private MySqlCommand comando;
        // ResultSet de Java
        private MySqlDataReader lector;
        public int insertar(Area area)
        {
            int resultado = 0;
            try
            {
                con = new MySqlConnection(DBManager.cadenaConexion);
                con.Open();
                // comando = con.CreateCommand();
                comando = new MySqlCommand();
                comando.Connection = con;
                comando.CommandType = CommandType.StoredProcedure;
                //comando.CommandText = "INSERT INTO area(nombre,activa)" +
                //    " VALUES('" + area.Nombre + "', 1)";
                comando.CommandText = "INSERTAR_AREA";

                comando.Parameters.Add("_id_area", MySqlDbType.Int32)
                    .Direction = ParameterDirection.Output;

                comando.Parameters.AddWithValue("_nombre", area.Nombre);

                resultado = comando.ExecuteNonQuery();

                area.IdArea = 
                    Int32.Parse(comando.Parameters["_id_area"].Value.ToString());
            } catch (Exception ex)
            {
                throw new Exception(ex.Message);
            } finally
            {
                try { con.Close(); } catch (Exception ex) { throw new Exception(ex.Message); }
            }
            return resultado;
        }

        public int modificar(Area area)
        {
            throw new NotImplementedException();
        }
        public int eliminar(int idArea)
        {
            throw new NotImplementedException();
        }


        public BindingList<Area> listaTodas()
        {
            BindingList<Area> areas = new BindingList<Area>();
            try
            {
                con = new MySqlConnection(DBManager.cadenaConexion);
                con.Open();
                // comando = con.CreateCommand();
                comando = new MySqlCommand();
                comando.Connection = con;
                comando.CommandType = CommandType.StoredProcedure;
                comando.CommandText = "LISTAR_AREAS_TODAS";

                lector = comando.ExecuteReader();
                while (lector.Read())
                {
                    Area area = new Area();
                    area.IdArea = lector.GetInt32("id_area");
                    area.Nombre = lector.GetString("nombre");
                    area.Activa = true;
                    areas.Add(area);
                }
                
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
            finally
            {
                try { con.Close(); } catch (Exception ex) { throw new Exception(ex.Message); }
            }
            return areas;
        }
    }
}
